tests/python_package_test/test_dask.py

```diff
@@ -656,6 +658,169 @@ def test_ranker(output, client, group):
     client.close(timeout=CLIENT_CLOSE_TIMEOUT)
 
 
+@pytest.mark.parametrize('task', tasks)
+@pytest.mark.parametrize('output', data_output)
+@pytest.mark.parametrize('eval_sizes', [[0.5, 1, 1.5], [0]])
+@pytest.mark.parametrize('eval_names_prefix', ['specified', None])
+def test_eval_set_no_early_stopping(task, output, eval_sizes, eval_names_prefix, client):
```

Comment:
so, guess what! I tested tonight and it seems like custom eval metrics work with the changes in this PR!

```python
import dask.array as da
import lightgbm as lgb
import numpy as np
from sklearn.datasets import make_regression
from distributed import Client, LocalCluster, wait

cluster = LocalCluster(n_workers=1)
client = Client(cluster)

# make sure you're starting with clean workers every time
client.restart()

# tightly control sizes of chunks, just for reproducibility purposes
X_np, y_np = make_regression(n_samples=1000, n_features=10)
X = da.from_array(X_np, chunks=(100, 10))
y = da.from_array(y_np, chunks=(100,))

# create an evaluation set with a single partitions
X_e = da.random.random((100, 10))
y_e = da.random.random((100,))

print(X_e.npartitions)

def _constant_metric(preds, train_data):
    metric_name = 'constant_metric'
    value = 0.708
    is_higher_better = False
    return metric_name, value, is_higher_better

model = lgb.DaskLGBMRegressor(
    num_iterations=5,
)

model.fit(
    X,
    y,
    eval_metric=[_constant_metric, "rmse"],
    eval_set=[
        (X_e, y_e)
    ]
)

print(model.evals_result_)
```

> {'valid_0': OrderedDict([('rmse', [15.892091924783728, 31.0336596895493, 44.319468550176666, 54.646394595034906, 64.42749393477636]), ('l2', [252.55858574577616, 963.0880337267573, 1964.2152925700984, 2986.228442236261, 4150.901974715644]), ('constant_metric', [0.708, 0.708, 0.708, 0.708, 0.708])])}

I know the point of this PR wasn't to fix https://github.com/microsoft/LightGBM/issues/3956 and usually I'd say "save that for another PR", but I think we can use the fact that metrics functions work to test the correctness of `evals_result_`.

Could you add one test that uses a custom metric function for each of classification, regression, and ranking? For now, it's ok to have that just use `"array"` (not the other data types), so this would add only 3 total tests. In that test, you could then check that `evals_result_` has exactly the values you expect.

Reply:
Haha cool sure no problem.

Reply:
Resolved in f9f1fc26

Reply:
Resolved in f9f1fc26